### PR TITLE
chore: refactor actions

### PR DESCRIPTION
**Pull request by brodrigu**
Refactored actions to attempt to use fewer resources.

Total processing time for all actions is actually greater (5.5 minute vs 4.4minutes), but time on Mac/Windows is reduced which in theory is a more expensive resource.

I don't feel strongly about merging this, but its a good case study.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -41,10 +59,32 @@ jobs:
         env:
           CI: true
         run: npm ci
-      - name: Lint
-        run: npm run lint
       - name: Test
         run: npm run test:unit
+
+  integration:
+    needs: lint
```
Comment (iamnewton) [inline .github/workflows/test.yml:66]: does it really need linting to work?  I feel like this is redundant given its run on `unit` testing as well

**Review by iamnewton — COMMENTED** [review]
Pretty sure we need to be on `node-version` 13

File: .github/workflows/test.yml (hunk 1 above)
Comment (brodrigu) [reply .github/workflows/test.yml:66]: I guess not, we could make these parallel

**Comment by brodrigu** [comment]
the integration tests execute the code on chrome, so the version of node for those only really matters in that it needs to support the features used by the test framework. node 12 seems to work, but I'm fine moving it to 13 if you prefer this.

**Review by iamnewton — COMMENTED** [review]
So is the idea here to:

1. separate out the linting and testing tasks individually
2. only run the linting on `ubuntu` and no where else?
3. run unit tests on all nodes
4. add in integration tests

I'm okay with this, but I do have some general suggestions.

For the unit tests, I think we should ditch the matrix as well and only unit test on ubuntu.  The integrations tests running on all makes perfect sense, but I don't think we need it for linting and unit tests.

**Pull request merged**